packages/snaps-sdk/src/jsx/validation.ts

```diff
@@ -76,6 +78,22 @@ export const ElementStruct: Describe<GenericSnapElement> = object({
   key: nullable(KeyStruct),
 });
 
+/**
+ * A struct for the {@link NonEmptyArray} type.
+ *
+ * @param struct - The struct for the non-empty array type.
+ * @returns The struct for the non-empty array type.
+ */
+function nonEmptyArray<Type, Schema>(
+  struct: Struct<Type, Schema>,
+): Struct<NonEmptyArray<Type>, any> {
+  return refine(
+    array(struct),
+    'NonEmptyArray',
+    (value) => value.length > 0,
+  ) as unknown as Struct<NonEmptyArray<Type>, Schema>;
```

Comment:
Could we just use `nonempty` from superstruct?

Reply:
Yes, I didn't know that was a thing.